### ISSUE TITLE
Usage of commit-and-tag-version within a CI/CD pipeline

### ISSUE THREAD

**Issue by olaurendeau** [absolute-version/commit-and-tag-version#204]
Hi,
I'm planning to use `commit-and-tag-version` within my CI/CD pipeline to automate it when a release is triggered.
Do you have recommendations for setting it up ? I've scan the docs but didn't found any reference to CI/CD.
Thanks !

**Comment by TimothyJones** [comment]
Good point! We really should add something on this in the documentation.

It depends a bit on your workflow, but you can see an example of an automated workflow [here](https://github.com/pact-foundation/pact-js/blob/master/scripts/ci/release.sh). That script is triggered by a repository dispatch - and has a slight disadvantage in that there's a race between the commit and push it does and anyone else merging - which might not work for everyone.

I usually make the release after pushing the tag - but it can make more sense to do it the other way around - where the changelog is merged, and _then_ the tag is generated.
